index.js

```diff
@@ -92,18 +92,41 @@ function realClickInternal(name, options, parent, window, renderer, done) {
 
 // Utilities
 
-function actionOnElementCenter(actionName) {
-  return function(selector, done) {
+function actionOnElement(actionName) {
+  return function(selector /*[, position], done */) {
+    const done = arguments[arguments.length - 1]
+    const position = arguments.length > 2 ? arguments[1] : undefined;
     if (typeof selector !== 'string') {
       return done(new TypeError(`${actionName}: "selector" must be a string`));
     }
-    debug(`Finding "${selector}"`);
     var child = this.child;
-    getBounds(this, selector).then(center).then(function(point) {
+
+    debug(`Finding "${selector}"`);
+    getBounds(this, selector).then(function calcPosition(bounds) {
+      if (position) {
+        if (typeof position.x !== 'number' || typeof position.y !== 'number') {
+          throw new TypeError('x and y must be numbers')
```

Comment:
Do you mind moving this type check up to the same place where we check the type of `selector`? Seems better to stop early if we can.